ui/lib/CancelStream.tsx

```diff
@@ -1,7 +1,48 @@
-import * as React from "react";
+import React, { useState } from "react";
+import { Address, stringToHex } from "viem";
+import StreamManager from "../../sdk/js/index";
 
-const CancelStream = () => {
-  return <div />;
+interface CancelStreamProps {
+  streamId: number;
+  reason: string;
+  creator: Address;
+  sm: StreamManager;
+}
+
+const CancelStream: React.FC<CancelStreamProps> = (props) => {
+  const [result, setResult] = useState<string | null>(null);
+  const hexReason = stringToHex(props.reason);
+
+  // console.log(props.streamId);
+  // console.log(hexReason);
+  // console.log(props.creator);
+  // console.log(props.sm);
+
+  const handleCancel = async () => {
+    try {
+      const result = await props.sm.cancel(
+        props.streamId,
+        hexReason,
+        props.creator
+      );
+      setResult(`Stream cancelled. Transaction Hash: ${result}`);
```

Comment:
issue: The issue with that tx talked about in tg is that a stream cannot be cancelled until `MIN_STREAM_LIFE` has been exceeded for the stream.

https://github.com/ApeWorX/ApePay/blob/61ad4b17996723b5a806aa21400a33b32c8d28af/contracts/StreamManager.vy#L31

It's probably a good idea to check `stream_is_cancelable` before allowing the user to waste their money here.

https://github.com/ApeWorX/ApePay/blob/61ad4b17996723b5a806aa21400a33b32c8d28af/contracts/StreamManager.vy#L210-L213

Reply:
> It's probably a good idea to check `stream_is_cancelable` before allowing the user to waste their money here.

This could probably raise an error in the SDK function `Stream.addFunds(amount)` (mimicking [python SDK](https://github.com/ApeWorX/ApePay/blob/main/sdk/py/apepay/__init__.py#L448) call to [`StreamManager.add_funds`](https://github.com/ApeWorX/ApePay/blob/main/contracts/StreamManager.vy#L195))